### ISSUE TITLE
Make Spring IDE unusable out of the box

### ISSUE THREAD

**Issue by amalic** [spring-projects/sts4#505]
Spring-IDE needs unnecessary configuration after installation. E.g. Maven inexes are not checked to be updated on startup. Final static fields which will be initialized by Spring's DI show an error message.

**Comment by kdvolder** [comment]
> E.g. Maven indexes are not checked to be updated on startup

Not everybody wants this as it poses a big drain on system resources. So that's why this is not enabled by default. If you like the feature enough that you don't mind the startup delay, and extra network and CPU load then you have to opt-in to this by enabling it. 

> Final static fields which will be initialized by Spring's DI show an error message.

Can you elaborate? I think you are probably talking about this:

https://bugs.eclipse.org/bugs/show_bug.cgi?id=547451

Correct?

Please be precise on which setting you would like STS to override the Eclipse default value for? And we can consider doing that if it seems reasonable (which I'd say in this case, it probably is).

**Comment by amalic** [comment]
What I mean by that is, that a developer should not need to start searching and reading through tons of documentation to fix your samples when using Spring IDE. They should run out of the box.

It's not like that I am setting up a spring develooment environment once per week. 

Take a look at how easy it is with Node development.
- install node
- install VSCode and start
- open the checked out project folder
- run: npm install
- run: npm start

-> works

Here's how it should work wit Spring-IDE on a fresh machine
- install java if not already installed
- install (extract) Spring IDE and start
- import existing maven project
- run: mvn spring-boot:run

**Comment by BoykoAlex** [comment]
Hmm... in the Eclipse/STS IDE world ideally after the Maven project imported you'd click on Run as Spring Boot application... or you can find it in the **Boot Dashboard** view, select and run it.
If there are errors on the Maven Java project after import in the IDE, right click Maven -> Update Project... (might be be M2E quirk). If there are errors on the pom.xml it might be about lifecycle mappings which you may install as a quick fix for the error.

No reason to look for prehistoric projects on baelung...

There are two wizards:
1. New Spring Boot project (File -> New -> Spring Starter Project). Generates new Spring Boot project in your workspace either Maven or Gradle and lets you pick "starters" (initial dependencies) for the project
2. Example Spring Boot project (File -> New -> Import Spring Getting Started Content). Allows to import example project in the workspace. Pick **spring-petclinic** - large example project

I'd disagree that creating and running Spring Boot project is more difficult than with a Node project... wasn't my first experience when i started with Spring.
Have you worked with Eclipse before? If you are a VSCode person maybe try to stay in VSCode and install Spring Boot extension for VSCode or perhaps the whole Spring Boot pack as it'd come with Spring Initializr wizard for new projects.

Let us know if this helps.

**Comment by kdvolder** [comment]
@amalic I agree that it should be a goal to have an as painless as possible getting started experience with sample code. The scenario you describe with maven for the most part should work just fine. I.e. you import such a project as a 'Existing maven project' using Eclipse project import wizard. Then after the project is imported in theory it should be setup correctly (dependencies etc are defined by the maven pom). 

Now Eclpse maven tooling does have its quirks and I'm sure there are lots of cases where you don't get that nice experience. This may or may not be something we can fix, depending on the specifics of the problem.

So If it doesn't work as expected, then it is something we can look at, but you will need to provide a concrete explicit example. I.e. if you tried something and it doesn't work nicely, document precisely what you did, which code/project did you import and how, and explain where it seems to go wrong, then we can try this for ourselves and analyze why it goes wrong and whether it is something that can be fixed (either by us or by Eclipse m2e). 

A general 'I wish it would just work for everything out of the box' isn't something we can do a lot about unfortunately. While we do agree with that sentiment of course, there's just not anything concrete enough to fix there.

**Comment by martinlippert** [comment]
@amalic As @BoykoAlex described there are plenty of ways to get started with your Spring project in the Spring Tools 4 for Eclipse environment that work out-of-the-box pretty much as you described as the desired way to get started. You don't even need to find or clone a git repo or run maven on the command line. Just hit the wizard to import a guide (for example) and press "run" afterwards - and you have a running and up-to-date Spring Boot project in your workspace in seconds.

Beyond that, we would of course be happy to improve the experience for additional scenarios and issues that people find. So if you have anything concrete (like a concrete project that doesn't work well after importing into the IDE), let us know and we would be more than happy to take a look.

**Comment by amalic** [comment]
@kdvolder Unfortunately this is not the case. At least not on Ubuntu Linux. I had to do additional configuration steps to make it work properly.

**Comment by martinlippert** [comment]
@amalic Let us see what exactly goes wrong when running down a simple starter path as a first step:

- download and install Spring Tools 4 for Eclipse
- run the IDE
- go to "New -> Import Spring Getting Started Content"
- select the "Rest Service" guide, uncheck "initial", and press finish
- expand the "local" tree in the boot dashboard view
- select the `gs-rest-service-complete` entry and press "run"

Does this simple example work out-of-the-box?
If not, which one of these steps do not work? And if they do not work, what is the error message or problem?

You said:

> What I mean by that is, that a developer should not need to start searching and reading through tons of documentation to fix your samples when using Spring IDE. They should run out of the box.

I really would like to get to the problems here to see what goes wrong with the examples and improve. So lets take a look at the concrete examples and start with the one that I use many times. And please point me to additional example projects that do not work out-of-the-box, as you mentioned.

**Comment by martinlippert** [comment]
Would be great to get some additional feedback on the previous comment.

**Comment by martinlippert** [comment]
Closing this one. Right now, you can download the latest Spring Tools for Eclipse disrtribution and just run that, you don't need to pre-install a JDK/JRE. Once open, you can create sample code by importing the guides via the provided wizard, just a few clicks away.
